src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -122,6 +122,7 @@ IFACEMETHODIMP CPowerRenameManager::AddItem(_In_ IPowerRenameItem* pItem)
         if (m_renameItems.find(id) == m_renameItems.end())
         {
             m_renameItems[id] = pItem;
+            m_isVisible.push_back(true);
```

Comment:
If the indices of added elements are consecutive and start from zero, consider changing `m_renameItems` to a vector. `std::advance(it, index);` in the function above this one is painfully slow with maps and results in quadratic performance.

Reply:
I don't see a reason to keep items in map either but I wasn't sure to add the change into this PR. 
`GetItemByIndex()` currently runs in `O(N)` but seems like it can be done in constant time.
`GetVisibleItemByIndex()` can be optimized to `O(logN)` after that.

There are some open issues and changes need to be made related to performance issues atm. Maybe we can focus on the matter on another issue/PR, or is it better to do it now @enricogior ?

Reply:
@muratakburak 
feel free to open an issue to track the perf issue and fix it in another PR.